cloudigrade/account/tests/test_views.py

```diff
@@ -396,3 +399,195 @@ def test_list_instances_as_superuser_with_bad_filter(self):
+        self.event3 = \
+            account_helper.generate_single_aws_instance_event(
+                instance=self.instance3, powered_time=powered_time)
+        self.event4 = \
+            account_helper.generate_single_aws_instance_event(
+                instance=self.instance4, powered_time=powered_time)
+        self.factory = APIRequestFactory()
+
+    def assertResponseHasEventData(self, response, event):
+        """Assert the response has data matching the instance event object."""
+        self.assertEqual(
+            response.data['id'], event.id
+        )
+        self.assertEqual(
+            response.data['instance'],
+            f'http://testserver/api/v1/instance/{event.instance.id}/'
+        )
+        self.assertEqual(
+            response.data['resourcetype'], event.__class__.__name__
+        )
+
+        if isinstance(event, AwsInstanceEvent):
+            self.assertEqual(
+                response.data['subnet'], event.subnet
+            )
+            self.assertEqual(
+                response.data['ec2_ami_id'], event.ec2_ami_id
+            )
+            self.assertEqual(
+                response.data['event_type'], event.event_type
+            )
+            self.assertEqual(
+                response.data['instance_type'], event.instance_type
+            )
+
+    def get_event_get_response(self, user, event_id):
+        """
+        Generate a response for a get-retrieve on the InstanceViewSet.
+
+        Args:
+            user (User): Django auth user performing the request
+            event_id (int): the id of the instance event to retrieve
+            data (dict): optional data to use as query params
+
+        Returns:
+            Response: the generated response for this request
+
+        """
+        request = self.factory.get('/event/')
+        force_authenticate(request, user=user)
+        view = InstanceEventViewSet.as_view(actions={'get': 'retrieve'})
+        response = view(request, pk=event_id)
+        return response
+
+    def get_event_list_response(self, user, data=None):
+        """
+        Generate a response for a get-list on the InstanceEventViewSet.
+
+        Args:
+            user (User): Django auth user performing the request
+            data (dict): optional data to use as query params
+
+        Returns:
+            Response: the generated response for this request
+
+        """
+        request = self.factory.get('/event/', data)
+        force_authenticate(request, user=user)
+        view = InstanceEventViewSet.as_view(actions={'get': 'list'})
+        response = view(request)
+        return response
+
+    def get_event_ids_from_list_response(self, response):
+        """
+        Get the instance event id values from the paginated response.
+
+        Args:
+            response (Response): Django response object to inspect
+
+        Returns:
+            set[int]: the instance id values found in the response
```

Comment:
`s/instance/event/`

Reply:
I need to pay more attention to my docstrings 🤦‍♀️ .